src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/ChangeDirectionDueDateNotificationHandler.java

```diff
@@ -0,0 +1,94 @@
+package uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.handlers.presubmit;
+
+import static java.util.Objects.requireNonNull;
+import static uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.AsylumCaseDefinition.DIRECTION_EDIT_DATE_DUE;
+import static uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.AsylumCaseDefinition.DIRECTION_EDIT_EXPLANATION;
+import static uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.AsylumCaseDefinition.HOME_OFFICE_CHANGE_DIRECTION_DUE_DATE_INSTRUCT_STATUS;
+import static uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.MessageType.DEFAULT;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.HomeOfficeInstruct;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.handlers.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.service.HomeOfficeInstructService;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.service.NotificationsHelper;
+
+@Slf4j
+@Component
+public class ChangeDirectionDueDateNotificationHandler implements PreSubmitCallbackHandler<AsylumCase> {
+
+    private HomeOfficeInstructService homeOfficeInstructService;
+    private NotificationsHelper notificationsHelper;
+
+    public ChangeDirectionDueDateNotificationHandler(
+        HomeOfficeInstructService homeOfficeInstructService,
+        NotificationsHelper notificationsHelper
+    ) {
+        this.homeOfficeInstructService = homeOfficeInstructService;
+        this.notificationsHelper = notificationsHelper;
+    }
+
+    @Override
+    public boolean canHandle(PreSubmitCallbackStage callbackStage, Callback<AsylumCase> callback) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+               && callback.getEvent() == Event.CHANGE_DIRECTION_DUE_DATE;
```

Comment:
Do you need state check here?

Reply:
State check added

Reply:
Hi @Lee365 I can not find the state check here. Am I missing anything?

Reply:
Good spot - Added state check now.